middleware/dnstap/setup.go

```diff
@@ -0,0 +1,43 @@
+package dnstap
+
+import (
+	"github.com/coredns/coredns/core/dnsserver"
+	"github.com/coredns/coredns/middleware"
+	lib "github.com/dnstap/golang-dnstap"
+	"github.com/mholt/caddy"
+	"github.com/pkg/errors"
+)
+
+func init() {
+	caddy.RegisterPlugin("dnstap", caddy.Plugin{
+		ServerType: "dns",
+		Action:     setup,
+	})
+}
+func setup(c *caddy.Controller) error {
+	c.Next() // 'dnstap'
+	if c.NextArg() {
+		return middleware.Error("dnstap", c.ArgErr())
+	}
+
+	tap := Dnstap{}
+	out, err := lib.NewFrameStreamOutputFromFilename("/tmp/db")
```

Comment:
I wonder how useful this is as opposed to a socket. Also this name should be configurable as having a default name that a process with root access creates can lead to security issues.